src/kubernetes.cc

```diff
@@ -400,29 +389,30 @@ KubernetesReader::GetPodAndContainerMetadata(
   const json::Array* container_statuses =
       status->Get<json::Array>("containerStatuses");
 
-  // Move the container specs into a map from name to spec.
-  std::map<std::string, const json::Object*> container_spec_by_name;
-  for (const json::value& c_spec : *container_specs) {
-    const json::Object* container_spec = c_spec->As<json::Object>();
-    const std::string name = container_spec->Get<json::String>("name");
-    container_spec_by_name.emplace(name, container_spec);
+  // Move the container statuses into a map from name to status.
+  std::map<std::string, const json::Object*> container_status_by_name;
+  for (const json::value& c_status : *container_statuses) {
+    const json::Object* container_status = c_status->As<json::Object>();
+    const std::string name = container_status->Get<json::String>("name");
+    container_status_by_name.emplace(name, container_status);
   }
 
-  for (const json::value& c_status : *container_statuses) {
+  for (const json::value& c_spec : *container_specs) {
     if (config_.VerboseLogging()) {
-      LOG(INFO) << "Container: " << *c_status;
+      LOG(INFO) << "Container: " << *c_spec;
     }
-    const json::Object* container_status = c_status->As<json::Object>();
-    const std::string name = container_status->Get<json::String>("name");
-    auto spec_it = container_spec_by_name.find(name);
-    if (spec_it == container_spec_by_name.end()) {
-      LOG(ERROR) << "Internal error; spec not found for container " << name;
-      continue;
+    const json::Object* container_spec = c_spec->As<json::Object>();
+    const std::string name = container_spec->Get<json::String>("name");
+    auto status_it = container_status_by_name.find(name);
+    const json::Object* container_status;
+    if (status_it == container_status_by_name.end()) {
+      container_status = nullptr;
+    } else {
+      container_status = status_it->second;
     }
-    const json::Object* container_spec = spec_it->second;
     result.emplace_back(GetLegacyResource(pod, name));
     result.emplace_back(
-        GetContainerMetadata(pod, container_status, container_spec,
+        GetContainerMetadata(pod, container_spec, container_status,
```

Comment:
Nit: generally not sure why you reversed the args here, but whatever the order ends up being, it'd be nice if it lined up with the definitions above. That way the code reads consistently throughout. Either reverse the order of the params, or the order of the definitions.

Reply:
It does, though -- `container_spec` is now declared in the loop header, and `container_status` is retrieved by name later. What am I missing?

Reply:
I didn't zoom out and realize this is being invoked inside of the loop. This looks good now that I better understand.